From a8457a33e309e84a9c3f04d406f15ed0a9ea1aec Mon Sep 17 00:00:00 2001
From: ysiyer <yegnesh.s.iyer@intel.com>
Date: Wed, 4 May 2016 16:13:14 +0530
Subject: [PATCH] EM/dc_xpwr_charger: Setting charging enabled upon enabling
 charger

Issue:
XPWR charger driver implements charging enable/disable upon setting
charger enable/disable power supply property. This was done since
disabling charger would affect further usb charger plug/unplug events.

As a result of this, while complying to usb spec, if the inlimit is
less than 100mA, the charging is disabled. This sequence is followed
when the current setting is 2mA. Subsequently when 500mA inlimit is set,
the charger's properties checked in power supply framework remains
unchanged and hence, does not trigger charging algorithm. As a result
while connecting SDP, sometimes the device shows "Not Charging" state.

Solution:
By setting charging enabled, when the property charger is enabled, upon
changing the inlimt from 2mA to 500mA, the charger property is shown as
changed and hence will trigger the framework to follow charging algorithm.
This solves the issue of "Not Charging" while the inlimit is changed from
2mA to 500mA.

Change-Id: I9a8ab22614ac386de336cd73174c81e7f671185e
Signed-off-by: ysiyer <yegnesh.s.iyer@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-30860
Reviewed-on: https://android.intel.com:443/502894
---
 drivers/power/dc_xpwr_charger.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/power/dc_xpwr_charger.c b/drivers/power/dc_xpwr_charger.c
index b385007..1d9da0f 100644
--- a/drivers/power/dc_xpwr_charger.c
+++ b/drivers/power/dc_xpwr_charger.c
@@ -605,7 +605,8 @@ static int pmic_chrg_usb_set_property(struct power_supply *psy,
 		ret = pmic_chrg_enable_charging(info, val->intval);
 		if (ret < 0)
 			dev_warn(&info->pdev->dev, "enable charger failed\n");
-		 info->is_charger_enabled = val->intval;
+		info->is_charging_enabled = val->intval;
+		info->is_charger_enabled = val->intval;
 		break;
 	case POWER_SUPPLY_PROP_CHARGE_CURRENT:
 		ret = pmic_chrg_set_cc(info, val->intval);
-- 
1.9.1

